src/app/user/default-organization-form/default-organization-form.component.ts

```diff
@@ -0,0 +1,123 @@
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DefaultOrganizationFormComponent implements OnInit, OnDestroy {
+  form!: FormGroup;
+  faSave = faSave;
+  saving = false;
+  organizationSelectItems: SelectItem[] = [];
+  defaultOrganizationRefControl = new FormControl();
+  faSitemap = faSitemap;
+
+  private handleActionsSubscription?: Subscription;
+
+  constructor(
+    private identityService: IdentityService,
+    private kubernetesClientService: KubernetesClientService,
+    private formBuilder: FormBuilder,
+    private store: Store,
+    private actions: Actions,
+    private router: Router,
+    private activatedRoute: ActivatedRoute,
+    private changeDetectorRef: ChangeDetectorRef,
+    private messageService: MessageService
+  ) {}
+
+  ngOnInit(): void {
+    this.form = new FormGroup({
+      defaultOrganizationRef: this.defaultOrganizationRefControl,
+    });
+    this.subscribeToUser();
+    this.handleActions();
+  }
+
+  private loadOrganizations(organizationList: OrganizationList, user: Entity<User | null>): void {
+    const getOrganizationMembersRequests = organizationList.items.map((organization) =>
+      this.kubernetesClientService.getOrganizationMembers(organization.metadata.name)
+    );
+
+    forkJoin(getOrganizationMembersRequests).subscribe((members) => {
+      const username = this.identityService.getUsername();
+      this.organizationSelectItems = organizationList.items
+        .filter((o, index) => (members[index].spec.userRefs ?? []).map((userRef) => userRef.name).includes(username))
+        .map(
+          (o) =>
+            ({
+              value: o.metadata.name,
+              label: o.spec.displayName ? `${o.spec.displayName} (${o.metadata.name})` : o.metadata.name,
+            } as SelectItem)
+        );
+
+      const organization = this.organizationSelectItems.find(
+        (organization) => organization.value === user.value?.spec.preferences?.defaultOrganizationRef
+      );
+      if (organization) {
+        this.defaultOrganizationRefControl.setValue(organization);
+      }
+    });
+  }
+
+  private subscribeToUser(): void {
+    // eslint-disable-next-line ngrx/no-store-subscription
+    this.store.select(selectUser).subscribe((user) => {
+      this.kubernetesClientService
+        .getOrganizationList()
+        .subscribe((organizationList) => this.loadOrganizations(organizationList, user));
+    });
+  }
+
+  save(): void {
+    if (this.form.valid) {
+      this.saving = true;
+      this.store.dispatch(
+        saveUserPreferences({
+          defaultOrganizationRef: this.form.value.defaultOrganizationRef?.value ?? null,
+        })
+      );
+    }
+  }
+
+  ngOnDestroy(): void {
+    this.handleActionsSubscription?.unsubscribe();
+  }
+
+  private handleActions(): void {
+    this.handleActionsSubscription = this.actions
+      .pipe(ofType(saveUserPreferencesSuccess, saveUserPreferencesFailure))
```

Comment:
Bräuchte es nicht ein take(1) ?

Reply:
I implemented it analogous to the `handleActions()` in `OrganizationFormComponent`. `take(1)` would work as well, because the action `saveUserPreferencesSuccess` and `saveUserPreferencesFailure` is thrown only once. But if it were thrown multiple times, it would not be wrong to listen to it every time and handle it accordingly.